tests/robot-tests/tests/libs/utilities.py

```diff
@@ -305,4 +277,27 @@ def remove_substring_from_right_of_string(string, substring):
 
 def user_clicks_element_if_exists(selector):
     if element_finder.find(selector, required=False) is not None:
-        sl.click_element(selector)
\ No newline at end of file
+        sl.click_element(selector)
+
+
+def __normalise_child_locator(parent_locator: object, child_locator: str) -> str:
+    if isinstance(parent_locator, str):
+        return child_locator
+    elif isinstance(parent_locator, WebElement):
+        # the below substitution is necessary if the parent is a Selenium WebElement in order to correctly find the 
+        # parent's descendants.  Without the preceding dot, the double forward slash breaks out of the parent container
+        # and returns the xpath query to the root of the DOM, leading to false positives or incorrectly found DOM 
+        # elements.
+        return child_locator.replace("xpath://", "xpath:.//")
```

Comment:
I know this is an existing issue before your change, but I think it's possible to have a string selector without the `xpath:` prefix, as any locator without the prefix defaults to xpath. So I think we need to regex replace "^//" with ".//" too

Reply:
Cool, good catch.  Added the following substitution, which yields the following results:

```
>>> re.sub(r'^(xpath:)?//', ".//", "asdf")
'asdf'
>>> re.sub(r'^(xpath:)?//', ".//", "xpath:hello")
'xpath:hello'
>>> re.sub(r'^(xpath:)?//', ".//", "xpath://hello")
'.//hello'
>>> re.sub(r'^(xpath:)?//', ".//", "//hello")
'.//hello
```